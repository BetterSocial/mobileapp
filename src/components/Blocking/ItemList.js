import * as React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

import Icon from 'react-native-vector-icons/FontAwesome';

import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const ItemList = ({label, active, onSelect, id}) => {
  const [type, setType] = React.useState('add');
  const onActive = () => {
    if (type === 'add') {
      onSelect(id, label, type);
      setType('remove');
    } else {
      onSelect(id, label, type);
      setType('add');
    }
  };
  return (
    <TouchableOpacity testID="click" style={styles.container} onPress={() => onActive()}>
      <Text style={styles.label}>{label}</Text>
      {active && <Icon testID="active-icon" name="check-circle" size={20} color={colors.red} />}
    </TouchableOpacity>
  );
};

export default ItemList;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 18,
    borderBottomColor: COLORS.alto,
    borderBottomWidth: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingRight: 20
  },
  label: {
    fontFamily: fonts.inter[400],
    fontSize: 14,
    paddingHorizontal: 20,
    color: COLORS.black,
    flex: 1
  }
});
