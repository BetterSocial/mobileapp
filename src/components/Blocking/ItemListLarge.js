import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import IconEn from 'react-native-vector-icons/Entypo';
import Handcuffs from '../../assets/icon-svg/handcuffs.svg';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
import IconFA5 from 'react-native-vector-icons/FontAwesome5';
const ItemListLarge = ({id, label, desc, iconReght, icon, onPress}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.header}>
        {icon === 'block' && <IconEn name="block" size={17} color={'#000'} />}
        {icon === 'handcuffs' && <Handcuffs />}
        <Text style={styles.label}>{label}</Text>
      </View>
      <View style={{flexDirection: 'row'}}>
        <Text style={styles.desc}>{desc}</Text>
        {iconReght && (
          <TouchableOpacity style={styles.btn}>
            <IconFA5 name="chevron-right" size={17} color={'#000'} />
          </TouchableOpacity>
        )}
      </View>
    </TouchableOpacity>
  );
};

export default ItemListLarge;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 15,
    paddingHorizontal: 22,
    borderTopColor: '#E0E0E0',
    borderTopWidth: 1,
  },
  btn: {
    justifyContent: 'center',
    alignItems: 'flex-end',
    width: 20,
    marginLeft: 15,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    marginLeft: 15,
    fontFamily: fonts.inter[500],
    color: '#000',
    fontSize: 14,
  },
  desc: {
    fontFamily: fonts.inter[400],
    fontSize: 12,
    color: colors.gray,
    marginTop: 5,
    flex: 1,
  },
});
