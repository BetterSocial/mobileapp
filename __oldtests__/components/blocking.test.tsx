import * as React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ItemList from '../../src/components/Blocking/ItemList';
import ItemListLarge from '../../src/components/Blocking/ItemListLarge';

configure({adapter: new Adapter()});
describe('component blocking', () => {
  it('ItemList snapshot', () => {
    const component = shallow(
      <ItemList
        onSelect={() => {}}
        active={true}
        label="The account is fake or not human"
        id={1}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('ItemList renders correctly', () => {
    const tree = renderer
      .create(
        <ItemList
          onSelect={() => {}}
          active={true}
          label="The account is fake or not human"
          id={1}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('ItemListLarge snapshot', () => {
    const component = shallow(
      <ItemListLarge
        onPress={() => {}}
        label="Block anonim indefinitely"
        id={1}
        desc="You will not be able to see each other’s posts, or message each other. The user’s reach will be reduced across BetterSocial, in particular for this post."
        icon="block"
        iconReght={false}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('ItemListLarge renders correctly', () => {
    const tree = renderer
      .create(
        <ItemListLarge
          onPress={() => {}}
          label="Block anonim indefinitely"
          id={1}
          desc="You will not be able to see each other’s posts, or message each other. The user’s reach will be reduced across BetterSocial, in particular for this post."
          icon="block"
          iconReght={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
