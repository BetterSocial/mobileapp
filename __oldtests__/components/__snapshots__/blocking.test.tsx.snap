// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component blocking ItemList renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingRight": 20,
      "paddingVertical": 18,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontFamily": "Inter-Regular",
        "fontSize": 14,
        "paddingHorizontal": 20,
      }
    }
  >
    The account is fake or not human
  </Text>
  <Icon
    color="#FF2E63"
    name="check-circle"
    size={20}
  />
</View>
`;

exports[`component blocking ItemList snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingRight": 20,
      "paddingVertical": 18,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontFamily": "Inter-Regular",
        "fontSize": 14,
        "paddingHorizontal": 20,
      }
    }
  >
    The account is fake or not human
  </Text>
  <Icon
    color="#FF2E63"
    name="check-circle"
    size={20}
  />
</ForwardRef>
`;

exports[`component blocking ItemListLarge renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderTopColor": "#E0E0E0",
      "borderTopWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 22,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#000"
      name="block"
      size={17}
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "marginLeft": 15,
        }
      }
    >
      Block anonim indefinitely
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#828282",
          "flex": 1,
          "fontFamily": "Inter-Regular",
          "fontSize": 12,
          "marginTop": 5,
        }
      }
    >
      You will not be able to see each other’s posts, or message each other. The user’s reach will be reduced across BetterSocial, in particular for this post.
    </Text>
  </View>
</View>
`;

exports[`component blocking ItemListLarge snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "borderTopColor": "#E0E0E0",
      "borderTopWidth": 1,
      "paddingHorizontal": 22,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Icon
      color="#000"
      name="block"
      size={17}
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "marginLeft": 15,
        }
      }
    >
      Block anonim indefinitely
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#828282",
          "flex": 1,
          "fontFamily": "Inter-Regular",
          "fontSize": 12,
          "marginTop": 5,
        }
      }
    >
      You will not be able to see each other’s posts, or message each other. The user’s reach will be reduced across BetterSocial, in particular for this post.
    </Text>
  </View>
</ForwardRef>
`;
